packages/jest-circus/src/legacy-code-todo-rewrite/jestAdapterInit.ts

```diff
@@ -201,7 +201,9 @@ export const runAndTransformResultsToJestFormat = async ({
         numFailingTests += 1;
       } else {
         status = 'passed';
-        numPassingTests += 1;
+        if (!testResult.phasing) {
```

Comment:
right, this is weird (like you mention). Can you say why this is needed? DO we generate test results now when there are no tests?

Reply:
Without it, all runs see two more passing "tests". These phasing results aren't tests, so I don't think anyone would expect that.

Default reporter shows it like:
```
  ● Stack Trace › does not print a stack trace for errors when --noStackTrace is given

    expect(received).toMatchSnapshot()

    Snapshot name: `Stack Trace does not print a stack trace for errors when --noStackTrace is given 1`

    - Snapshot  - 1
    + Received  + 1

      Test Suites: 1 failed, 1 total
    - Tests:       3 failed, 3 total
    + Tests:       3 failed, 2 passed, 5 total
      Snapshots:   0 total
      Time:        <<REPLACED>>
      Ran all test suites matching /testError.test.js/i.
```

Tests for custom reporters also have the same bug:
```
  ● Custom Reporters Integration › IncompleteReporter for flexibility

    expect(received).toMatchSnapshot()

    Snapshot name: `Custom Reporters Integration IncompleteReporter for flexibility 1`

    - Snapshot  - 2
    + Received  + 2

      onRunComplete is called
    - Passed Tests: 1
    + Passed Tests: 3
      Failed Tests: 0
    - Total Tests: 1
    + Total Tests: 3
```

I feel that these numbers shouldn't be coming out of circus, and should be derived by `@jest/test-result#addResult`, but I didn't look into why that isn't happening, and I strongly suspect that's also a massive change.

Reply:
what phasing results are not from a test? if there is setup and teardown stuff not connected to a test, we should move that outside the results array (somehow).

I might have to check this out locally and play with it to understand what's going on 😅 I feel like my questions sounds dumb, so there's something I'm missing here